extensions/ql-vscode/src/pure/log-summary-parser.ts

```diff
@@ -0,0 +1,42 @@
+// REVIEW: Perhaps this file should be renamed now that it also includes the 
+// data model on top of the parsing logic?
+export interface EvaluatorLogData {
+    queryCausingWork: string;
+    predicateName: string;
+    millis: number;
+    resultSize: number;
+    ra?: Pipeline[];
+}
+  
+interface Pipeline {
+    // Key: pipeline identifier; Value: array of pipeline steps 
+    pipeline: Map<string, string[]>;
+}
+  
+/**
+ * A pure method that parses a string of evaluator log summaries into
+ * an array of EvaluatorLogData objects. 
+ * 
+ */
+ export function parseVisualizerData(logSummary: string): EvaluatorLogData[] {
+    // Remove newline delimiters because summary is in .jsonl format.
+    const jsonSummaryObjects: string[] = logSummary.split('\n\n');
```

Comment:
Try using:

```suggestion
    const jsonSummaryObjects: string[] = logSummary.split(os.EOL + os.EOL);
```

need to import `os` at the start of the file as well.